### ISSUE TITLE
StackOverflow when upgrading from DiffEqBase v6.90.0 to v6.91.2

### ISSUE THREAD

**Issue by charleskawczynski** [SciML/DiffEqBase.jl#779]
We're seeing a StackOverflow error when upgrading from DiffEqBase v6.90.0 to v6.91.2. Here is a [gist](https://gist.github.com/charleskawczynski/cd866e72e5f78987f99b7bf806cf3a59) of the error. I'll see if I can dig into this more, come up with a smaller example (or maybe someone will find out / know what's wrong). I'm just opening here to track.

**Comment by ChrisRackauckas** [comment]
Yes, it looks like the version of the ForwardDiff promotions that we released was not robust to having `undef` values in Structs, Sets, and AbstractArrays. That should be fixed in https://github.com/SciML/DiffEqBase.jl/pull/778 and I'll do a patch release there.
